pkgs/applications/blockchains/solana/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, rustPlatform
+, IOKit
+, Security
+, AppKit
+, openssl
+, pkg-config
+, rustfmt
```

Comment:
It needs `udev`, `zlib`, and also `rocksdb` for the validator, although for just the `solana` CLI, `rocksdb` may not be needed. Currently this does not build on Linux:

```
builder for '/nix/store/6bpwzj3k3864vcv3x9q559f0gzqas9q9-solana-testnet-cli-1.9.2.drv' failed with exit code 101; last 10 log lines:
    thread 'main' panicked at 'Unable to find libudev: `"pkg-config" "--libs" "--cflags" "libudev"` did not exit successfully: exit status: 1
    --- stderr
    Package libudev was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libudev.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libudev' found
    ', /build/solana-testnet-cli-1.9.2-vendor.tar.gz/hidapi/build.rs:55:54
    note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  warning: build failed, waiting for other jobs to finish...
  error: build failed
```

Reply:
Let me add those.

I tried to run the checkphase. But on darwin, it fails with a missing framework that doesn't exist on nixos. So we won't be able to run the checkphase ever on darwin (most probably). I left the checkphase on linux to run in case someone can test. If you find out missing dependencies for the checkphase (most probably rocksdb for example) just let me know and I can add those to the checkInputs. If we find that in the end we still can't run the tests, we can always leave the checkphase to false.

Reply:
I ran into some issues with the tests in https://github.com/NixOS/nixpkgs/pull/121009, but that was for the full test suite, not limited to `cli`. A few of those issues have been fixed in the meantime, but most haven’t. I can try and build this, maybe the tests for just `cli` are actually fine.